logging/google/cloud/logging/handlers/transports/background_thread.py

```diff
@@ -152,9 +151,10 @@ class BackgroundThreadTransport(Transport):
     def __init__(self, client, name):
         http = copy.deepcopy(client.connection.http)
         http = client.connection.credentials.authorize(http)
-        self.client = Client(client.project,
-                             client.connection.credentials,
-                             http)
+        # __class__ avoids a circular import
+        # https://github.com/GoogleCloudPlatform/google-cloud-python/issues/2160
```

Comment:
Well, the client needs to access this, but this needs to make a copy of the client, which is why the circular dependency exists.